app/src/main/java/co/develhope/chooseyouowncocktail_g2/adapter/DrinkCardAdapter.kt

```diff
@@ -52,47 +37,44 @@ class DrinkCardAdapter(
                 binding.drinkName.text = this.name
 
                 binding.drinkShortDescription.text = this.shortDescription
+                showPreferiteRecycleView(this, holder)
 
                 binding.drinkImage.setImageByUrl(
                     "https://www.thecocktaildb.com/images/media/drink/metwgh1606770327.jpg",
                     60,
                     60
                 )
 
-
                 binding.buttonGoToDetail.setOnClickListener {
                     action(DrinkAction.GotoDetail(this.id))
                 }
-                // context.resources.getIdentifier("ic_fav_on", "drawable", context.packageName)
-
-
-                var switch = false
-                val bf = binding.drinkFavourite
-
-                val imOn =
-                    context.resources.getIdentifier("ic_fav_off", "drawable", context.packageName)
-                val imOff =
-                    context.resources.getIdentifier("ic_fav_on", "drawable", context.packageName)
-                bf.setOnClickListener {
-                    if (!switch) {
-                        bf.setBackgroundResource(imOn)
-                        switch = true
-                    } else {
-                        bf.setBackgroundResource(imOff)
-                        switch = false
-                    }
-                }
+
+               binding.drinkFavourite.setOnClickListener {
+                   this.favourite = !this.favourite
+                   action(DrinkAction.SetPref(this, this.favourite))
+                   showPreferiteRecycleView(this, holder)
```

Comment:
Perchè passare l'older? 
Potevi fare `private fun showPreferiteRecycleView(drinkCard: Drink) : Int{
        var icon = 0
        if (drinkCard.favourite){
           icon= R.drawable.ic_fav_on
        }else{
           icon= R.drawable.ic_fav_off
        }
        return icon
    }`

` showPreferiteRecycleView(this)`